backend/assignments/1/src/main/java/cryptotradingplatform/Main.java

```diff
@@ -0,0 +1,251 @@
+package cryptotradingplatform;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import javafx.util.Pair;
+import org.json.simple.JSONArray;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+public class Main {
+
+    private static TradingPlatform tradingPlatform;
+
+    public static JsonNode parseJsonFile(String transactionPath) throws IOException {
+//        String transactionPath = Main.class.getClassLoader().getResource("small_transaction.json").getPath();
+        JSONParser jsonParser = new JSONParser();
+        JSONArray jsonArray = null;
+        try {
+            jsonArray = (JSONArray) jsonParser.parse(new FileReader(transactionPath));
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+
+        ObjectMapper objectMapper = new ObjectMapper();
+            return objectMapper.readTree(jsonArray.toString());
+    }
+
+    public static ArrayList<String[]> parseCSV(Path path) throws IOException {
+
+        ArrayList<String[]> parsedCSV = new ArrayList<String[]>();
+        String line = "";
+        String splitBy = ",";
+//            String tradersCSVPath = Main.class.getClassLoader().getResource("coins.csv").getPath();
+            //parsing a CSV file into BufferedReader class constructor
+            BufferedReader br = Files.newBufferedReader(path);
+            br.readLine();
+
+            while ((line = br.readLine()) != null)   //returns a Boolean value
+            {
+                String[] coinProperties = line.split(splitBy);    // use comma as separator
+                    System.out.println("Coins [Rank=" + coinProperties[1] + ", Name=" + coinProperties[2] + ", Symbol=" + coinProperties[3] + ", Price=" + coinProperties[4] + "]");
+//                Coin coin = new Coin(Integer.parseInt(coinProperties[1]), coinProperties[2], coinProperties[3], Double.parseDouble(coinProperties[4]), Long.parseLong(coinProperties[5]));
```

Comment:
Override the `toString()` method in the coin object instead of writing this here.